OpenRA.Mods.Common/Traits/KillsSelf.cs

```diff
@@ -18,13 +18,24 @@ class KillsSelfInfo : ConditionalTraitInfo
 		[Desc("Remove the actor from the world (and destroy it) instead of killing it.")]
 		public readonly bool RemoveInstead = false;
 
-		public override object Create(ActorInitializer init) { return new KillsSelf(this); }
+		[Desc("The amount of time (in ticks) before the actor dies. Two values indicate a range between which a random value is chosen.")]
+		public readonly int[] Delay = { 250 };
+
+		public override object Create(ActorInitializer init) { return new KillsSelf(init.Self, this); }
 	}
 
-	class KillsSelf : ConditionalTrait<KillsSelfInfo>, INotifyAddedToWorld
+	class KillsSelf : ConditionalTrait<KillsSelfInfo>, INotifyAddedToWorld, ITick
 	{
-		public KillsSelf(KillsSelfInfo info)
-			: base(info) { }
+		int lifetime;
+
+		public KillsSelf(Actor self, KillsSelfInfo info)
+			: base(info)
+		{
+			if (info.Delay.Length > 1)
+				lifetime = self.World.SharedRandom.Next(info.Delay[0], info.Delay[1]);
+			else
+				lifetime = info.Delay[0];
```

Comment:
This will crash if `info.Delay.Length == 0`.  You could perhaps move https://github.com/OpenRA/OpenRA/pull/12672/files#diff-1d9a691633bee72d36992c8598cd3656R92 somewhere reusable and then use it here?